### DISCUSSION TITLE
WMS map doensn't open in QField, but does in desktop QGIS

### DISCUSSION THREAD

**Discussion by Dutch5972** [opengisch/QField#4722] [Bugs]
Hi,
Since a year I am using a WMS backround map in QField. For using this map I have a subscribtion with its developer. Till recently it worked fine on both QGIS and QField (Samsung Android). But suddenly it stopped working in QField although it is still working in QGIS. The provider mentioned no other customers reported any problems with using this background map.
I reïnstalled the latest QGIS version (Lima), the latest Qfield sync plug-in and the newest version of QField. The 'Save path' setting in project properties is set in 'relative'.
In QField in the layerinfo it says that the layer is invalid, which might be due to a network problem, an unknown file or wrong configuration of the project. But the project settings have not changed and on QGIS the WMS layer is working properly. Other standard WMS background maps and aerial photos are working fine though.

Does anyone know how to tackle this issue? Thanks in advance!

**Comment by jbell6** [comment]
Did you work out a solution to this? I have the same problem.

**Reply by nirvn** [reply]
@Dutch5972 , @jbell6 , this will require a test project to replicate the issue. I've not encountered WMS layers that didn't work in QField.

**Reply by Dutch5972** [reply]
Hi,

With the latest release of QField the problem seemed to be solved but a short while later the WMS map couldn’t be loaded anymore.
What I do to work around it, is to install an older version of QField in the phone/tablet (QField 2.xx). I am running QField 2.8.4-Insightful Indri on my phone and with that the map is loading as it should. You can download the older versions of QField for free through websites like Aptoide. Make sure you switch off the automatic updates in the QField app settings on your device.
Hopefully newer releases of QField will tackle this issue more properly, so also the new features of QField can be used.

Good luck!

**Comment by Dutch5972** [comment]
Hi,

 

With the latest release of QField the problem seemed to be solved but a short while later the WMS map couldn’t be loaded anymore.

What I do to work around it, is to install an older version of QField in the phone/tablet (QField 2.xx). I am running QField 2.8.4-Insightful Indri on my phone and with that the map is loading as it should. You can download the older versions of QField for free through websites like Aptoide. Make sure you switch off the automatic updates in the QField app settings on your device.

Hopefully newer releases of QField will tackle this issue more properly, so also the new features of QField can be used.

 

Good luck!

 

 

Aernout Theunissen

 

 

Van: jbell6 ***@***.***> 
Verzonden: woensdag 31 juli 2024 08:56
Aan: opengisch/QField ***@***.***>
CC: Dutch5972 ***@***.***>; Author ***@***.***>
Onderwerp: Re: [opengisch/QField] WMS map doensn't open in QField, but does in desktop QGIS (Discussion #4722)

 

Did you work out a solution to this? I have the same problem.

**Reply by nirvn** [reply]
@Dutch5972 , hopefully a user will be able to provide a sample project to replicate the issue, a necessary first step to fix this. 

As mentioned above, if you are able to provide such a test, it'd be super useful.

**Comment by kiri03** [comment]
I have a project in QGIS where everything is displayed correctly, including several WMS layers, but when I send that project to QField there is a specific layer that is not displayed and an error related to the SSL certificate appears:

Map request failed [error: SSL handshake failed: The issuer certificate of a locally looked up certificate could not be found url: https://ovc.catastro.meh.es/Cartografia/WMS/ServidorWMS.aspx?

I have deleted all the layers leaving only the layer with the WMS layer and still nothing is displayed in QField, the same error appears.
I attach the QGZ file with the WMS layer to be able to reproduce the error in QField and see if someone can provide me with a solution.)
[Catastro_Spain.zip](https://github.com/user-attachments/files/16442732/Catastro_Spain.zip)

**Comment by Dutch5972** [comment]
Hi,

 

Please find attached a QField test project to investigate the issue. The WMS TopoPlus-map is displayed properly in QGIS, but doesn’t load in the newer QField application (3.xx). I tested it on QField 3.3.1 Darién and it cannot be displayed. In the older versions of QField (until 2.xx) it used to work well. 

 

Regards,

 

 

Aernout Theunissen

 

 

Van: Mathieu Pellerin ***@***.***> 
Verzonden: woensdag 31 juli 2024 15:13
Aan: opengisch/QField ***@***.***>
CC: Dutch5972 ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [opengisch/QField] WMS map doensn't open in QField, but does in desktop QGIS (Discussion #4722)

 

@Dutch5972 <https://github.com/Dutch5972>  , hopefully a user will be able to provide a sample project to replicate the issue, a necessary first step to fix this.

As mentioned above, if you are able to provide such a test, it'd be super useful.

**Reply by nirvn** [reply]
@Dutch5972 , thanks for taking the time to provide a sample project. It has however not reached GitHub yet as attachments to emails are not added in discussion comments. You'll need to add the sample project directly on GitHub using your web browser. 

Cheers.

**Reply by kiri03** [reply]
Here we have a test project.
It only has a layer of the Spanish cadastre. In QGIS the layer is displayed correctly but in QField nothing is displayed and the error appears:

_Map request failed [error: SSL handshake failed: The issuer certificate of a locally looked up certificate could not be found url: https://ovc.catastro.meh.es/Cartografia/WMS/ServidorWMS.aspx?_


[Catastro_Spain.zip](https://github.com/user-attachments/files/16456391/Catastro_Spain.zip)

**Reply by nirvn** [reply]
@kiri03 , yeah, I've tested your project in the issue you filed and added some comments. In that case, it appears to be a certificate issue with the server itself. I'm hoping @Dutch5972 's project might lead us to a different issue.

**Comment by Dutch5972** [comment]
[Test_QField.zip](https://github.com/user-attachments/files/16456593/Test_QField.zip)

Please find a test project for QField to investigate the issue. The WMS TopoPlus map is working properly in QGISbut not in the QField-versions 3.xx any longer.

**Comment by Dutch5972** [comment]
Just added the attachment through GitHub.

 

Good luck!

 

Van: Mathieu Pellerin ***@***.***> 
Verzonden: donderdag 1 augustus 2024 14:24
Aan: opengisch/QField ***@***.***>
CC: Dutch5972 ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [opengisch/QField] WMS map doensn't open in QField, but does in desktop QGIS (Discussion #4722)

 

@Dutch5972 <https://github.com/Dutch5972>  , thanks for taking the time to provide a sample project. It has however not reached GitHub yet as attachments to emails are not added in discussion comments. You'll need to add the sample project directly on GitHub using your web browser.

Cheers.

**Comment by nirvn** [comment]
@Dutch5972 , I've tried your project. The layer didn't load at first as the server doesn't appear to play well when trying to connect from outside of the Netherlands. But as soon as I switched my VPN on and tunneled through Netherlands, the layer loaded on QField (Android and Linux) just fine:

![image](https://github.com/user-attachments/assets/093b92fb-148e-4d13-b1d5-9ff5bf015e52)

I'm using the latest version of QField, namely 3.3.8.

**Reply by Dutch5972** [reply]
![Screenshot_error message_TopoPlus_QField 3 3 8](https://github.com/user-attachments/assets/d30f0a79-1527-4d9b-a95e-9129869fffcb)

Hi, just updated to QField 3.3.8 but still the same error message occurs (see attachment). The map still cannot be loded on my Samsung tablet.

**Reply by jbell6** [reply]
> @Dutch5972 , I've tried your project. The layer didn't load at first as the server doesn't appear to play well when trying to connect from outside of the Netherlands. But as soon as I switched my VPN on and tunneled through Netherlands, the layer loaded on QField (Android and Linux) just fine:
> 
> ![image](https://private-user-images.githubusercontent.com/1728657/354235621-093b92fb-148e-4d13-b1d5-9ff5bf015e52.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI1MjI0MjgsIm5iZiI6MTcyMjUyMjEyOCwicGF0aCI6Ii8xNzI4NjU3LzM1NDIzNTYyMS0wOTNiOTJmYi0xNDhlLTRkMTMtYjFkNS05ZmY1YmYwMTVlNTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MDFUMTQyMjA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2M0ZmYxZTdjNDE5OGI3OWQzY2QyY2FiZDgyZGRmOWM0Nzk4Zjg5N2UyOWM2OGJmNmMyOWU0NjUyZWFiZGU3YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Q-WMbc_JTKC73V8DQ4_Gq6szfsoASMOoi9dWbSsQxG0)
> 
> I'm using the latest version of QField, namely 3.3.8.

My issue also appears to be solved by using a VPN.

**Comment by Dutch5972** [comment]
I just downloaded the 3.3.8 version of QField, but still the same issue here. I sent you error message through GitHub.

 

 

Van: Mathieu Pellerin ***@***.***> 
Verzonden: donderdag 1 augustus 2024 15:04
Aan: opengisch/QField ***@***.***>
CC: Dutch5972 ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [opengisch/QField] WMS map doensn't open in QField, but does in desktop QGIS (Discussion #4722)

 

 <https://github.com/Dutch5972> @Dutch5972 , I've tried your project. The layer didn't load at first as the server doesn't appear to play well when trying to connect from outside of the Netherlands. But as soon as I switched my VPN on and tunneled through Netherlands, the layer loaded on QField (Android and Linux) just fine:

 <https://github.com/user-attachments/assets/093b92fb-148e-4d13-b1d5-9ff5bf015e52> image.png (view on web)

I'm using the latest version of QField, namely 3.3.8.

**Comment by Dutch5972** [comment]
Still it is strange that in the older (2.xx) versions of QField the WMS map is working properly, while in the newer releases (3.xx) it is not. Is this an issue that can be fixed in future releases of QField?

 

 

Van: jbell6 ***@***.***> 
Verzonden: vrijdag 2 augustus 2024 11:10
Aan: opengisch/QField ***@***.***>
CC: Dutch5972 ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [opengisch/QField] WMS map doensn't open in QField, but does in desktop QGIS (Discussion #4722)

 

@Dutch5972 <https://github.com/Dutch5972>  , I've tried your project. The layer didn't load at first as the server doesn't appear to play well when trying to connect from outside of the Netherlands. But as soon as I switched my VPN on and tunneled through Netherlands, the layer loaded on QField (Android and Linux) just fine:



I'm using the latest version of QField, namely 3.3.8.

My issue also appears to be solved by using a VPN.
